### DISCUSSION TITLE
Starting, pausing, resuming and stopping the charging

### DISCUSSION THREAD

**Discussion by phulstaert** [leeyuentuen/alfen_wallbox#149] [Q&A]
When creating an automation, it is reasonable to expect to do a few basic operations.
I will try to summarize what I want to do.

1. I want to identify the car that is connected to the Alfen Wallbox. Depending on the car, I might want to do different things.
I do really mean the car (serial number or something like that), not the card that was hold next to the Alfen Wallbox.

2. Is there a way to use HA to push some data to the Alfen Wallbox to emulate a RFID card being held to the Wallbox?
If I plug in my car at home, it would be wonderful if I could recognise the car and just start charging without holding my card against the Alfen Wallbox. You could also use this to stop the charging session and restart it, without going outside and unplugging and plugging it back in.

3. Charging starts automatically when you plug in, but sometimes you need to pause the charging. I was under the impression that putting the Wallbox to 0A would stop the charging session, putting it to 1A-5A would pause the session and between 6A-32A would charge.
The issue is that when I put in in 5A and try resuming after a minute, this works, but when I wait a longer period of time, it will not resume the charging. I don't know if it is the Wallbox that refuses or if it is the car itself.
_I tried putting the charging current to 6A every half hour for a minute, to keep the connection "alive", but this was too long and wouldn't resume. I am trying each 15 minutes since today._
So the question here is
- How do I "pause" a session correctly (0W to the car) and how long can I leave it paused before it cancels the session?
- How can I resume a session after pausing (>= 6A to the car)
- How can I start a session after sending a RFID token to the Alfen Wallbox trough an automation?

Also, Why can't we charge at 1A-5A? Is this a certain regulation of the EU or something like that? Because I CAN set my car to limit the charging to 1A-5A. (but I want a solution that works on every car)
If it is a certain law, did I misunderstood that the "green charging" does enable charging under 6A?

I know these are a lot of questions, but I find them extremely important (and I think they also are very useful for new users)

**Comment by zimbo86** [comment]
Hi, anything new on that? Receiving my EV in january and want to achieve the same..

**Reply by phulstaert** [reply]
From what I can tell:

- Putting the device on 0A will stop the charging. To resume, you have to scan your RFID card again (I don't know what this does on a configuration without a card)
- Putting the device on 1-5A (I only tested 5A), it will pause the charging. So you can resume without scanning an RFID card, but some cars (not the charger) can "timeout". I have a Tesla and notice that it often refuses to resume when being idle around 30 minutes.
The solution I made for this, was enabling charging at 6A when the time has 0 or 30 minutes. This would still timeout sometimes. So I added when the time has 0, 1, 30 or 31 minutes and now it always works. (When changing the Amperage, it sometimes takes more than 1 minute for the car to respond; I guess it dit not always charged when I changed it for a single minute.)
- Charging between 6-32A works as expected, but always with a latency of about a minute. So changing the current each minute might be overkill. (I do this however and never have any issues).

If you need some more information or tips or tricks about smart charging, feel free to ask questions.

**Reply by zimbo86** [reply]
really appreciate your input, thank you!
One basic point is actually not clear to me:
Can I set solar charging mode to comfort and change the power via comfort level even when there is no smart meter connected to it?
![image](https://github.com/user-attachments/assets/796400b1-970e-482c-9fae-5caad4fc4414)

**Reply by phulstaert** [reply]
As far as I know, the P1 smart meter is a requirement for the solar charging.

This is how it works (as far as I know - I could be wrong)

**Solar charging**
Alfen looks at the amount of power returned to the grid or drawn from the grid. It will adjust the amount of power going to the car, so all solar is used and maximum the "solar comfort level" is drawn from the grid.
This has a HUGE drawback for people with a home battery. This home battery will supply power from the moment you draw power from the grid. The car will empty the home battery very quickly.

**Socket max power**
You can decide how much power goes to the car. Since you use Home Assistant, you can recreate the solar charging, but also integrate smart features.
ex: I have 2 automatisations. 

1. **Solar charging**: An offset of the SoC (state of charge) of my home battery is the percentage on how much energy from my solar goes to my car. Difficult sentence, but this is for example:
Home battery is 20%; we add 20 (and get 40%); If the solar panels generate 4800W (this would be 20A on a 240V system); 40% of that goes to the car (or 8A) and 60% (the other half) is used to further charge the Home battery.
When the home battery is 70%; we add 20 (and get 90%); If the solar panels generate 4800W (or 20A on a 240V system); 90% of that goes to the car (or 18A) and the 10% (the other half) is used to further charge the Home battery.
=> The home battery gets charged first, but as it increases in SoC, more and more is going to the car. This has as a result there is no unneeded charging and discharging of the Home battery to increase its lifetime and the situation where the battery is full, but there is not enough solar to reach that 6A to start charging the car does not often occur.

2. **grid charging**: I look at what I draw from the net. I charge the car at a rate so I won't draw any more then 4000W. This because where I live (Belgium) they monitor my average draw from the grid each 15 minutes and the maximum of that month is my "peak usage" of that month. Then hey make an average of those peak usages of the last 12 months and I have to pay about 40€ for each 1000W above 2500W. So if I where to charge at 22kW; that is about 20x 40€ or 800€ extra each year.
4kW means 80€ each year and is something I get even when not charging the car, but just cooking, washing machine,... normal life :-)
This does however deplete the Home battery and is more for the winter months.

**Reply by pstoopen** [reply]
Hi, what is the easiest way to start, stop and resume charging ? Just by changing the current ? Do you have some examples of scripts ?

**Reply by phulstaert** [reply]
Yes, set current to 0A to stop; 1A to 5A (doesn't matter) to pauze and >= 6A to charge.

You do this with an action (you can test this in "developer tools" / "actions" and then select "set current limit") - this is also in the documentation
![image](https://github.com/user-attachments/assets/f49c0f46-07cc-4a64-9fc3-0e4300a36592)

edit: you can find the documentation on github here. Look at wiki and then for examples at "My automation for charging" from the menu on the right

**Comment by J1Buskens** [comment]
Did you ever find a solution for the RFID emulation?

**Reply by phulstaert** [reply]
No; I think it is impossible for the moment since it even isn't in the modbus registers...  ([link](https://alfen.com/file-download/download/public/1610))

**Reply by zimbo86** [reply]
I have to deal also with this, because the charging costs are handled by a accounting service provider.. 
My first idea was to build an ESP32 with a small DC motor that presents the RFID to the wallbox ^^
Will dig into it when my car arrives at the end of january

**Reply by phulstaert** [reply]
That would be an option, but i wonder if you look up the RFID reader inside the Alfen box, it probably communicates with I2C or serial or something like that. It would be easier to "write" something on that line i would guess...

[here](https://knowledge.alfen.com/download/attachments/293634103/UM_EVP_DE_2.4_EN_web.pdf) on page 11 it says

RFID card reader: 
RFID (NFC) ISO/IEC 14443A/B, MiFare Classic
13.56 MHz, DESFire
Maximum length: 7 bytes

**Reply by J1Buskens** [reply]
gettting off topic here, but for anyone looking for a simple solution if you just have one card: The easiest solution for me was to use the "Plug & Charge authorisation" mode instead of RFID and insert the ID of the card as the "Plug and Charge ID", the backoffice still recognises it as legit.

**Reply by phulstaert** [reply]
Good to know and a solution to most.

But in my case i don't think it is usable.
I have a RFID card form work to use with my car and only my car.
My charging station is publicly accessible.

I would rather have something like
- If my car is on the home wifi 
  and the signalstrength is between certain values (or: on the driveway)
  and no card was presented
  use the RFID from work.
  
I was wondering if the charger has any information about the connected car. A serial number or some sort of identification. That would be even more accurate than the rules above.
On public chargers you can see the SoC of the connected car, so there is information send to the charger. But even the SoC is not seen by Alfen. (or the plugin)

**Reply by zimbo86** [reply]
> gettting off topic here, but for anyone looking for a simple solution if you just have one card: The easiest solution for me was to use the "Plug & Charge authorisation" mode instead of RFID and insert the ID of the card as the "Plug and Charge ID", the backoffice still recognises it as legit.

perfect, thats all what I need and works perfectly. Can confirm that all transactions are properly recognized.

**Reply by zimbo86** [reply]
Hey @J1Buskens  and @phulstaert 
sorry for offtopic, but i ordered a second EV which is also loaded by my Alfen charger.
Currently i used your approach and entered the Pllug&Charge ID to charge without RFID and report all charging operations to backoffice:
<img width="341" height="183" alt="image" src="https://github.com/user-attachments/assets/27c8c9ec-785e-482f-8cfd-c9340f334c36" />
I now need to handle my work car (with backoffice) and my personal car, which shouldnt use backoffice.
As i dont got the car now, I remembered that if P&C is not set, the car wont load at all.
Do i need to create a dummy RFID or sth like that in ACE Service Installer?
May it is even possible to recognize which car is connected and assign the appropiate P&C ID on the fly before the loading starts?

**Reply by phulstaert** [reply]
I have "kind of" the same issue. I want to use the ChargeID of the company car on the company car, but the cars of the rest of my family that come and charge need to use either their own or a "free local" one of myself.

I had issues with this when writing this post (so a year ago - I don't know if it is still the same).
Changing an ID "during" charging does not work. So using a local "dummy" and checking if it is the case that both the charger and the company car are charging and the car is on the same wifi and if so, changing the chargingID, does not work.
Even when pausing the charging and resuming it (<5A charging OR even 0A charging).

The solution I use now is just holding my cell phone (with a charging card in the smartphone-cover) to the charger. It takes longer to plug the car in then to hold the card to the charger AND I usually have my phone in my hands when exiting the car.
I know it is not the "automatic" solution, but it works and is full proof...

**Comment by pstoopen** [comment]
Hi Peter,

 

What might be simpler to do, provided that your car supports it, is to integrate the car with Home Assistant. 

In HA, you can then see that the car is connected. With EV Smart Charging, you can create a number of additional conditions. In EV Smart Charging, you can also see that the car is connected.

The charger doesn’t give any information about the car.

 

Kind regards, groet,

Paul

 

 

Van: Peter Hulstaert ***@***.***>
Beantwoorden - Aan: leeyuentuen/alfen_wallbox ***@***.***>
Datum: maandag 20 januari 2025 om 11:13
Aan: leeyuentuen/alfen_wallbox ***@***.***>
CC: pstoopen ***@***.***>, Comment ***@***.***>
Onderwerp: Re: [leeyuentuen/alfen_wallbox] Starting, pausing, resuming and stopping the charging (Discussion #149)

 

Good to know and a solution to most.

But in my case i don't think it is usable.
I have a RFID card form work to use with my car and only my car.
My charging station is publicly accessible.

I would rather have something like
If my car is on the home wifi
and the signalstrength is between certain values (or: on the driveway)
and no card was presented
use the RFID from work.
I was wondering if the charger has any information about the connected car. A serial number or some sort of identification. That would be even more accurate than the rules above.
On public chargers you can see the SoC of the connected car, so there is information send to the charger. But even the SoC is not seen by Alfen. (or the plugin)

**Reply by zimbo86** [reply]
pretty cool informations here, will receive my EV tomorrow.
Sorry for offtopic, @pstoopen  do you know if theres sth similar to EV Smart Charging that is not based on flexible tarifs but on charging based on a home battery / set load automatically based on PV income kW?

**Comment by pstoopen** [comment]
Hi,

 

What kind of charger do you have ? An Alfen ? Most of the chargers has standard solar funcitionality.

 

Kind regards,

Paul

 

Van: zimbo86 ***@***.***>
Beantwoorden - Aan: leeyuentuen/alfen_wallbox ***@***.***>
Datum: maandag 20 januari 2025 om 14:30
Aan: leeyuentuen/alfen_wallbox ***@***.***>
CC: pstoopen ***@***.***>, Mention ***@***.***>
Onderwerp: Re: [leeyuentuen/alfen_wallbox] Starting, pausing, resuming and stopping the charging (Discussion #149)

 

pretty cool informations here, will receive my EV tomorrow.
Sorry for offtopic, @pstoopen do you know if theres sth similar to EV Smart Charging that is not based on flexible tarifs but on charging based on a home battery / set load automatically based on PV income kW?

**Reply by phulstaert** [reply]
This does not take into account you have a home battery. The solar functionality will increase the charging current to a point where you are pulling the allowed Wattage from the grid. If you have  a 10kW system with a home battery, you will be drawing 10kW out of that battery and deplete it in no time.

@zimbo86 I don't know if this is what you are looking for, but I made [a post](https://github.com/leeyuentuen/alfen_wallbox/discussions/43#discussioncomment-9981107) about something similar a while ago.

**Reply by zimbo86** [reply]
> This does not take into account you have a home battery. The solar functionality will increase the charging current to a point where you are pulling the allowed Wattage from the grid. If you have a 10kW system with a home battery, you will be drawing 10kW out of that battery and deplete it in no time.
> 
> @zimbo86 I don't know if this is what you are looking for, but I made [a post](https://github.com/leeyuentuen/alfen_wallbox/discussions/43#discussioncomment-9981107) about something similar a while ago.

i switched over to EVCC.. thats exactly what i was looking for :)
